bundles/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/api/EvohomeApiClient.java

```diff
@@ -139,33 +142,33 @@ private void setHeatingZoneOverride(String zoneId, HeatSetPoint heatSetPoint) th
         apiAccess.doAuthenticatedPut(url, heatSetPoint);
     }
 
-    private UserAccount requestUserAccount() throws TimeoutException {
+    private @Nullable UserAccount requestUserAccount() throws TimeoutException {
         String url = EvohomeApiConstants.URL_V2_BASE + EvohomeApiConstants.URL_V2_ACCOUNT;
         return apiAccess.doAuthenticatedGet(url, UserAccount.class);
     }
 
     private Locations requestLocations() throws TimeoutException {
-        Locations locations = new Locations();
-        if (useraccount != null) {
+        Locations locations = null;
```

Comment:
This was intentionally. Because `return apiAccess.doAuthenticatedGet(url, Locations.class);` returns nullable Location. A check is needed before returning it. So i also have to declare it before the if statement. If you have further suggestions, let me know.

Reply:
Ah, I missed that. It would be better if `doAuthenticatedGet` either returned non-null or threw an exception, but this would probably be outside scope of your PR.